uppsrc/CtrlCore/X11Ctrl.h

```diff
@@ -94,7 +94,9 @@
 
 	virtual void    EventProc(XWindow& w, XEvent *event);
 	virtual bool    HookProc(XEvent *event);
-	Window  GetWindow() const         { return top ? top->window : None; }
+	Window  GetWindow() const         { return top && utop ? utop->window : None; }  //aris002
+	static int    GetCaretBlinkTime()               { return 500; } //aris002
+    void SetCaret(int x, int y, int cx, int cy); //aris002
```

Comment:
SetCaret method should be completely removed it is no longer used in GTK and Win32 implementation.